lib/fps_counter.dart

```diff
@@ -1,39 +1,33 @@
-import 'dart:math' as math;
+import 'package:flutter/scheduler.dart';
 
 import 'game/game.dart';
 
+const _maxFrames = 60;
+const frameInterval =
+    Duration(microseconds: Duration.microsecondsPerSecond ~/ _maxFrames);
+
 mixin FPSCounter on Game {
-  /// List of deltas used in debug mode to calculate FPS
-  final List<double> _dts = [];
+  List<FrameTiming> _previousTimings = [];
+
+  @override
+  void onTimingsCallback(List<FrameTiming> timings) =>
+      _previousTimings = timings;
 
-  /// Returns whether this [Game] is should record fps or not
+  /// Returns whether this [Game] is should record fps or not.
   ///
   /// Returns `false` by default. Override to use the `fps` counter method.
   /// In recording fps, the [recordDt] method actually records every `dt` for statistics.
   /// Then, you can use the [fps] method to check the game FPS.
+  @Deprecated('Flame is now using Flutter frame times, will be removed in v1')
   bool recordFps();
 
-  /// This is a hook that comes from the RenderBox to allow recording of render times and statistics.
-  @override
-  void recordDt(double dt) {
-    if (recordFps()) {
-      _dts.add(dt);
-    }
-  }
-
-  /// Returns the average FPS for the last [average] measures.
-  ///
-  /// The values are only saved if in debug mode (override [recordFps] to use this).
-  /// Selects the last [average] dts, averages then, and returns the inverse value.
-  /// So it's technically updates per second, but the relation between updates and renders is 1:1.
-  /// Returns 0 if empty.
+  /// Returns the FPS based on the frame times from [onTimingsCallback].
   double fps([int average = 1]) {
-    final List<double> dts = _dts.sublist(math.max(0, _dts.length - average));
-    if (dts.isEmpty) {
-      return 0.0;
-    }
-    final double dtSum = dts.reduce((s, t) => s + t);
-    final double averageDt = dtSum / average;
-    return 1 / averageDt;
+    return _previousTimings.length *
+        _maxFrames /
+        _previousTimings.map((t) {
+          return (t.totalSpan.inMicroseconds ~/ frameInterval.inMicroseconds) +
+              1;
```

Comment:
I think the multiline is fine but this +1 hopefully fits above? seems odd to have a +1 alone in a line